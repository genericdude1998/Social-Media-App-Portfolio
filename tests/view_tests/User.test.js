import React from 'react'
import { shallow } from 'enzyme';
import ConnectedUser from '../../src/view/UsersInfo/User';
import {mockId, mockUser} from '../mockValues';

jest.spyOn(React, 'useEffect').mockImplementation((effect) => effect());
jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');
    return {
        __esModule:true,
        ...originalModule,
        useParams: jest.fn(() => {return{id: mockId}}),
    }
});

const mockOnFetchUser = jest.fn();

let wrapper = shallow(<ConnectedUser.WrappedComponent
    user={mockUser}
    onFetchUser={mockOnFetchUser}
/>);

describe('User', () => {
    it('should call OnFetchUser with expected params', () => {
        expect(mockOnFetchUser).toHaveBeenCalledWith(mockId);
    });
    it('should match snapshot if user is not undefined', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should match snapshot if list is undefined', () => {
        let wrapper = shallow(<ConnectedUser.WrappedComponent
            user={undefined}
            onFetchUser={mockOnFetchUser}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});
