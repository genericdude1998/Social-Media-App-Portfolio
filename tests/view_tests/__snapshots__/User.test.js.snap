// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should match snapshot if list is undefined 1`] = `
<div
  className="baseUser null"
>
  <h1>
    Loading
  </h1>
</div>
`;

exports[`User should match snapshot if theme dark 1`] = `
<div
  className="baseUser dark"
>
  <div
    className="baseUserContent dark"
  >
    <h1>
      mock_name
    </h1>
    <h3>
      City: 
      NY
    </h3>
    <h3>
      Joined: 
      mock_date
    </h3>
    <h3>
      Description:
    </h3>
    <p />
  </div>
</div>
`;

exports[`User should match snapshot if user is not undefined 1`] = `
<div
  className="baseUser null"
>
  <div
    className="baseUserContent null"
  >
    <h1>
      mock_name
    </h1>
    <h3>
      City: 
      NY
    </h3>
    <h3>
      Joined: 
      mock_date
    </h3>
    <h3>
      Description:
    </h3>
    <p />
  </div>
</div>
`;
